pkg/docker/config/config_test.go

```diff
@@ -886,3 +886,92 @@ func TestValidateKey(t *testing.T) {
 		assert.Equal(t, tc.isNamespaced, isNamespaced)
 	}
 }
+
+func TestSetGetCredentials(t *testing.T) {
+	const (
+		username = "username"
+		password = "password"
+	)
+
+	getAuth := func(sys *types.SystemContext, input string) types.DockerAuthConfig {
+		ref, err := reference.ParseNamed(input)
+		require.NoError(t, err)
+		auth, err := GetCredentialsForRef(sys, ref)
+		require.NoError(t, err)
+		return auth
+	}
+
+	// Unset the home dir to make the test not fallback to other auth files
+	// there.
+	tmpDir, err := ioutil.TempDir("", "auth-test-")
+	require.NoError(t, err)
+	os.Setenv("HOME", tmpDir)
```

Comment:
Doesn’t this have to be restored after the test? Also the comment above says “unset”, not “change”.

Either way, wouldn’t using `getCredentialsWithHomeDir` allow testing this without affecting $HOME at all?

Reply:
Yes, switched to use `getCredentialsWithHomeDir`